packages/attribute-registry-process/src/services/attributeRegistryService.ts

```diff
@@ -40,7 +41,7 @@ export function attributeRegistryServiceBuilder(
   return {
     async createDeclaredAttribute(
       apiDeclaredAttributeSeed: ApiDeclaredAttributeSeed,
-      authData: AuthData,
+      authData: AuthDataUI | AuthDataM2M,
```

Comment:
After a look at both propositions I find this one cleaner.

The only thing I would change is to call the asserts only when we actually need `AuthData` at `RegistryService` level instead on the Router.

For example in the following code I would define authData as following and make the proper assert in order to refine its type when needed

```suggestion
      authData: AuthData,
```

Reply:
Thanks for the feedback! After examining both proposals again, I agree with you: this is the one I like the most. 

I also like what you suggest in the comment: moving the assertions in the service would make explicit the authorized roles at the service level, where the business logic happens - now, the service acts knowing nothing about the role that the router authorizes and this can create issues (for example, it is theoretically possible to call the service calls with a wrong role in the tests: https://github.com/pagopa/interop-be-monorepo/pull/392#discussion_r1561036837).

If @galales  / @Viktor-K / @MalpenZibo  agree, I could close #305 and proceed with this approach.